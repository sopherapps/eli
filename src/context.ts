import React from "react";
import { UserAppConfig, Tab } from "./data/models";

export enum Theme {
  Dark,
  Light,
}

const defaultUserAppConfig: UserAppConfig = {
  tabs: {},
  tabOrder: [],
  createTab: () => ({
    id: `${new Date().getTime()}`,
    title: "",
    visualizations: {},
    order: [],
  }),
  updateTab: (id: string, tab: Tab) => tab,
  deleteTab: (id: string) => {},
};

export const ThemeContext = React.createContext(Theme.Dark); // other possibility is light
export const UserAppConfigContext = React.createContext(defaultUserAppConfig);
