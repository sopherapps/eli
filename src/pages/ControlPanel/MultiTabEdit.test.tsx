import React from "react";
import renderer from "react-test-renderer";
import { defaultUserAppConfig, UserAppConfigContext } from "../../context";
import MultiEdit from "./MultiTabEdit";

test("renders the MultiEdit screen correctly", () => {
  const appConfig = { ...defaultUserAppConfig };
  appConfig.tabs = {
    hi: { id: "hi", title: "hi you", visualizations: {}, order: [] },
    hello: { id: "hello", title: "hello ha", visualizations: {}, order: [] },
  };
  appConfig.tabOrder = ["hi", "hello"];

  const tree = renderer
    .create(
      <UserAppConfigContext.Provider value={appConfig}>
        <MultiEdit />
      </UserAppConfigContext.Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
