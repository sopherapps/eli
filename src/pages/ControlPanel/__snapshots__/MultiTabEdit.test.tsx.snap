// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MultiEdit screen correctly 1`] = `
<div
  className="container"
>
  <section
    className="page-banner d-flex justify-space-between"
  >
    <div
      className="page-heading"
    >
      <p
        className="subtitle"
      >
        TABS
      </p>
      <h4
        className="title"
      >
        Control Panel
      </h4>
    </div>
    <button
      className="btn btn-round"
      onClick={[Function]}
    >
      <img
        alt="Create New Tab"
        src="add_white.svg"
      />
    </button>
  </section>
  <section
    className="scrollview"
  >
    <div
      className="card"
    >
      <div
        className="card-header d-flex justify-space-between"
      >
        <input
          className="editable-text"
          data-name="title"
          onChange={[Function]}
          spellCheck={true}
          type="text"
          value="hi you"
        />
        <div
          className="card-control d-flex justify-space-between"
        >
          <button
            className="btn"
          >
            <img
              alt="edit"
              src="edit_white.svg"
            />
          </button>
          <button
            className="btn"
          >
            <img
              alt="close"
              src="close_white.svg"
            />
          </button>
        </div>
      </div>
      <div
        className="card-body list-with-images"
      />
    </div>
    <div
      className="card"
    >
      <div
        className="card-header d-flex justify-space-between"
      >
        <input
          className="editable-text"
          data-name="title"
          onChange={[Function]}
          spellCheck={true}
          type="text"
          value="hello ha"
        />
        <div
          className="card-control d-flex justify-space-between"
        >
          <button
            className="btn"
          >
            <img
              alt="edit"
              src="edit_white.svg"
            />
          </button>
          <button
            className="btn"
          >
            <img
              alt="close"
              src="close_white.svg"
            />
          </button>
        </div>
      </div>
      <div
        className="card-body list-with-images"
      />
    </div>
  </section>
</div>
`;
